cmd/nri-kubernetes/main.go

```diff
@@ -202,7 +206,7 @@ func runScrapers(c *config.Config, ksmScraper *ksm.Scraper, kubeletScraper *kube
 	return nil
 }
 
-func setupKSM(c *config.Config, clients *clusterClients) (*ksm.Scraper, error) {
+func setupKSM(c *config.Config, clients *clusterClients, namespaceCache *discovery.NamespaceInMemoryStore) (*ksm.Scraper, error) {
```

Comment:
Nit: now we need to add a new parameter to have a common cache, maybe it would make more sense to have your initial approach (directly adding `nsFilter` as an argument to `setupKSM` and `setupKubelet`)

Reply:
I prefer to do it like this, since the cache is always going to be vacuumed, even if there is no namespaceFilter, moving the logic outside the setupKSM and setupKubelet would make me create the check namespace != nil outside and an extra nsFilter != nil WDYT?